File: backend/app/utils/india_bounds.py
from app.core.config import get_settings

settings = get_settings()


def is_in_india(lat: float, lng: float) -> bool:
    return (
        settings.india_min_lat <= lat <= settings.india_max_lat
        and settings.india_min_lng <= lng <= settings.india_max_lng
    )


def validate_india_coords(lat: float, lng: float) -> None:
    if not is_in_india(lat, lng):
        raise ValueError(
            "Coordinates are outside India. "
            "Street Info Hub only covers Indian roads."
        )


INDIA_CENTER = {"lat": 20.5937, "lng": 78.9629}
INDIA_BOUNDS = {
    "min_lat": settings.india_min_lat,
    "max_lat": settings.india_max_lat,
    "min_lng": settings.india_min_lng,
    "max_lng": settings.india_max_lng,
}

File: frontend/src/utils/indiaBounds.js
export const INDIA_BOUNDS = {
  minLat: 6.5546,
  maxLat: 35.6745,
  minLng: 68.1766,
  maxLng: 97.4025,
}

export const INDIA_CENTER = [20.5937, 78.9629]
export const INDIA_DEFAULT_ZOOM = 5

// Leaflet LatLngBounds format [[sw_lat, sw_lng], [ne_lat, ne_lng]]
export const INDIA_LEAFLET_BOUNDS = [
  [INDIA_BOUNDS.minLat, INDIA_BOUNDS.minLng],
  [INDIA_BOUNDS.maxLat, INDIA_BOUNDS.maxLng],
]

export function isInIndia(lat, lng) {
  return (
    lat >= INDIA_BOUNDS.minLat &&
    lat <= INDIA_BOUNDS.maxLat &&
    lng >= INDIA_BOUNDS.minLng &&
    lng <= INDIA_BOUNDS.maxLng
  )
}
